src/sugar3/graphics/popwindow.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (C) 2016 Abhijit Patel
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+from gi.repository import Gtk
+from gi.repository import Gdk
+from gi.repository import GdkX11
+
+from sugar3.graphics import style
+from sugar3.graphics.toolbutton import ToolButton
+from sugar3.graphics.toolbarbox import ToolbarBox
+
+from jarabe.model import shell
+
+
+class PopWindow(Gtk.Window):
+    """
+    UI interface for activity Pop-up Windows.
+    PopWindows are the windows that open on the top of the current window.
+    These pop-up windows don't cover the whole screen.
+    They contain canvas content, alerts messages, a tray and a
+    toolbar.
+
+    Kwargs:
+        title (gettext.gettext or _() ): title label of the pop-up window
+        set_titlebox (bool): False if titlebox is not needed
+        window_xid (xlib.Window): xid of the parent window
+    """
+
+    def __init__(self, title=None, set_titlebox=True, window_xid=None):
+        Gtk.Window.__init__(self)
```

Comment:
Please proxy the kword argments to the constructor:

```
def __init__(self, title=None, set_titlebox=True, window_xid=None, **kwargs):
    Gtk.Window.__init__(self, **kwargs)
```

Since the GtkWindow is a GOBject, that lets me go:

```
p = PopWindow(transistent_for=self)
```

rather than

```
p = PopWindow()
p.props.transistent_for = self